### PR TITLE
chore(rln-relay): namespaced flags

### PR DESCRIPTION
**Pull request by rymnc**
Should resolve https://github.com/status-im/nwaku/issues/1196

This PR namespaces few flags that are exclusively used with rln-relay. This will 
unblock https://github.com/status-im/infra-role-nim-waku/pull/5 and 
https://github.com/status-im/infra-nim-waku/pull/61

### CODE REVIEW

File: examples/v2/config_chat2.nim [hunk 1]
```diff
@@ -270,24 +270,25 @@ type
...
       desc: "WebSocket address of an Ethereum testnet client e.g., ws://localhost:8540/",
       defaultValue: "ws://localhost:8540/"
-      name: "eth-client-address" }: string
+      name: "rln-relay-eth-client-address" }: string
     
     rlnRelayEthMemContractAddress* {.
       desc: "Address of membership contract on an Ethereum testnet", 
       defaultValue: ""
-      name: "eth-mem-contract-address" }: string
+      name: "rln-relay-eth-membership-contract-address" }: string
```
[the remark this answers is not in the document]
Comment (jakubgs) [reply examples/v2/config_chat2.nim:291]: Or the `membership` part. As far as I can tell there are no other contracts, so no need to distinguish between them.

[the remark this answers is not in the document]
Comment (rymnc) [reply examples/v2/config_chat2.nim:291]: Dropping `membership` in favour of `address`

[the remark this answers is not in the document]
Comment (rymnc) [reply examples/v2/config_chat2.nim]: Addressed in [3f3bc1e](https://github.com/status-im/nwaku/pull/1204/commits/3f3bc1e8a7681c1ae40de402ab6ca8cd0eb26140)

File: examples/v2/config_chat2.nim [hunk 2]
```diff
@@ -270,24 +270,25 @@ type
...
   
     rlnRelayEthAccount* {.
-      desc: "Ethereum account address for an Ethereum testnet", 
+      desc: "Ethereum account address for an Ethereum testnet",
+      # NOTE: This can be derived from the private key, but kept for future use
       defaultValue: ""
-      name: "eth-account-address" }: string
+      name: "rln-relay-eth-account-address" }: string
     
     rlnRelayEthAccountPrivKey* {.
```
Comment (jm-clius) [inline examples/v2/config_chat2.nim:278]:
```suggestion
    rlnRelayEthAccountPrivateKey* {.
```

For consistency the metric field should match the config `name`, except that `-` (dashes) are converted to `camelCase`

File: examples/v2/config_chat2.nim [hunk 3]
```diff
@@ -270,24 +270,25 @@ type
       name: "rln-relay-id-commitment" }: string
   
     rlnRelayEthAccount* {.
-      desc: "Ethereum account address for an Ethereum testnet", 
+      desc: "Ethereum account address for an Ethereum testnet",
+      # NOTE: This can be derived from the private key, but kept for future use
       defaultValue: ""
-      name: "eth-account-address" }: string
+      name: "rln-relay-eth-account-address" }: string
```
Comment (jm-clius) [inline examples/v2/config_chat2.nim:276]:
```suggestion
      name: "rln-relay-eth-account" }: string
```

To match the field `rlnRelayEthAccount` (or "Address" can be suffixed to the field)

File: examples/v2/config_chat2.nim [hunk 4]
```diff
@@ -270,24 +270,25 @@ type
...
     
     rlnRelayEthAccountPrivKey* {.
       desc: "Account private key for an Ethereum testnet",
       defaultValue: ""
-      name: "eth-account-privatekey" }: string
+      name: "rln-relay-eth-account-privatekey" }: string
```
Comment (jm-clius) [inline examples/v2/config_chat2.nim:281]:
```suggestion
      name: "rln-relay-eth-account-private-key" }: string
```
(Throughout most of the codebase "private key" is considered two words, though I'm sure it's up for debate :) )

File: waku/v2/node/config.nim [hunk 5]
```diff
@@ -155,23 +155,24 @@ type
   
     rlnRelayEthAccount* {.
       desc: "Account address for the Ethereum testnet Goerli", 
+      # NOTE: This can be derived from the private key, but kept for future use
       defaultValue: ""
-      name: "eth-account-address" }: string
+      name: "rln-relay-eth-account-address" }: string
```
Comment (jm-clius) [inline waku/v2/node/config.nim:160]:
```suggestion
      name: "rln-relay-eth-account" }: string
```

File: waku/v2/node/config.nim [hunk 6]
```diff
@@ -155,23 +155,24 @@ type
...
 
     rlnRelayEthAccountPrivKey* {.
```
Comment (jm-clius) [inline waku/v2/node/config.nim:162]:
```suggestion
    rlnRelayEthAccountPrivateKey* {.
```

File: waku/v2/node/config.nim [hunk 7]
```diff
@@ -155,23 +155,24 @@ type
...
       desc: "Account private key for the Ethereum testnet Goerli",
       defaultValue: ""
-      name: "eth-account-privatekey" }: string
+      name: "rln-relay-eth-account-privatekey" }: string
```
Comment (jm-clius) [inline waku/v2/node/config.nim:165]:
```suggestion
      name: "rln-relay-eth-account-private-key" }: string
```

**Review by jm-clius — APPROVED** [review]
I have added a couple of comments re consistency. In general, the `fieldName` for the config item should match the `--config-name`, but with dashes replaced with camelCase.

Once the consistency issue is addressed, no further comments from me. LGTM.

File: waku/v2/node/config.nim (hunk 7 above)
Comment (rymnc) [reply waku/v2/node/config.nim]: Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

File: waku/v2/node/config.nim (hunk 6 above)
Comment (rymnc) [reply waku/v2/node/config.nim]: Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

File: waku/v2/node/config.nim (hunk 5 above)
Comment (rymnc) [reply waku/v2/node/config.nim]: Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

File: examples/v2/config_chat2.nim (hunk 4 above)
Comment (rymnc) [reply examples/v2/config_chat2.nim]: Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

File: examples/v2/config_chat2.nim (hunk 3 above)
Comment (rymnc) [reply examples/v2/config_chat2.nim]: Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

File: examples/v2/config_chat2.nim (hunk 2 above)
Comment (rymnc) [reply examples/v2/config_chat2.nim]: Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

File: examples/v2/config_chat2.nim (hunk 3 above)
Comment (rymnc) [reply examples/v2/config_chat2.nim:276]: Chose to continue with the address suffix to match the other config items, updated the variable

**Review by staheri14 — APPROVED** [review]
Thanks, looks good to me!

**Comment by rymnc** [comment]
> > I have added a couple of comments re consistency. In general, the `fieldName` for the config item should match the `--config-name`, but with dashes replaced with camelCase.
> > Once the consistency issue is addressed, no further comments from me. LGTM.
> 
> I can still see some exceptions, but probably are fine like that @jm-clius

Addressed in [4cf046e](https://github.com/status-im/nwaku/pull/1204/commits/4cf046ebe3af84a4dcdde241bff080dfad54b2d2)

**Pull request merged**